pkg/operator/ceph/cluster/osd/deviceSet.go

```diff
@@ -293,3 +296,15 @@ func deviceSetPVCID(deviceSetName, pvcTemplateName string, setIndex int) string
 	deviceSetName = strings.Replace(deviceSetName, ".", "-", -1)
 	return fmt.Sprintf("%s-%s-%d", deviceSetName, cleanName, setIndex)
 }
+
+func createValidImageVersionLabel(image string) string {
+	// regex to replace characters used by image name format that are not allowed in label values
+	re := regexp.MustCompile("[/:]")
+	cephImageVersion := re.ReplaceAllString(image, "_")
```

Comment:
Instead of replacing the `/:` characters, what if we just take the substring after the `:`? We really just want the image tag, not the repo, right? For example, the label would just be `v17.2.6` instead of `quay.io/ceph/ceph:v17.2.6`.

Reply:
In case not using an official image, I think it's better to store the name of the image at the time the OSD was created in order to let user know what image has been used.
For example, I use self-build image like `<my company name>/ceph:<version>`

Reply:
The repo likely won't change over time, right? But that's ok to keep the full image name if you prefer.